src/Models/HydrostaticFreeSurfaceModels/mg_implicit_free_surface_solver.jl

```diff
@@ -56,17 +60,31 @@ step `Δt`, gravitational acceleration `g`, and free surface at the `n`-th time-
 """
 function MGImplicitFreeSurfaceSolver(grid::AbstractGrid, 
                                      settings = nothing,
-                                     gravitational_acceleration = nothing, 
+                                     gravitational_acceleration = g_Earth,
+                                     reduced_dim = (false, false, false),
                                      placeholder_timestep = -1.0)
     arch = architecture(grid)
 
+    right_hand_side = Field{Center, Center, Nothing}(grid)
+
     # Initialize vertically integrated lateral face areas
     ∫ᶻ_Axᶠᶜᶜ = Field{Face, Center, Nothing}(with_halo((3, 3, 1), grid))
     ∫ᶻ_Ayᶜᶠᶜ = Field{Center, Face, Nothing}(with_halo((3, 3, 1), grid))
 
     vertically_integrated_lateral_areas = (xᶠᶜᶜ = ∫ᶻ_Axᶠᶜᶜ, yᶜᶠᶜ = ∫ᶻ_Ayᶜᶠᶜ)
 
     compute_vertically_integrated_lateral_areas!(vertically_integrated_lateral_areas)
+    coeffs = compute_matrix_coefficients(vertically_integrated_lateral_areas, grid, gravitational_acceleration)
+    matrix_constructors, diagonal, problem_size = matrix_from_coefficients(arch, right_hand_side, coeffs, reduced_dim)  
```

Comment:
why second argument here is `right_hand_side`? where is `matrix_from_coefficients` defined?